src/wfs/src/main/java/org/vfny/geoserver/wfs/servlets/TestWfsPost.java

```diff
@@ -316,13 +316,20 @@ public String errorResponse(Exception e) {
     }
 
     String getProxyBaseURL() {
-        GeoServer geoServer = (GeoServer) GeoServerExtensions.bean("geoServer");
+        GeoServer geoServer = getGeoServer();
         if (geoServer != null) {
-            geoServer.getGlobal().getSettings().getProxyBaseUrl();
+            return geoServer.getGlobal().getSettings().getProxyBaseUrl();
         }
         return null;
     }
 
+    /*
+     * Protected for testing purposes.
+     */
+    protected GeoServer getGeoServer() {
```

Comment:
Minor suggestion, it might be worth using `@VisibleForTesting` annotation instead of a comment.